tcms/testcases/tests/test_views.py

```diff
@@ -564,3 +566,33 @@ def test_disabled_priority_now_shown(self):
         self.assertEqual(HTTPStatus.OK, response.status_code)
         self.assertContains(response, 'Set P3')
         self.assertNotContains(response, 'Set P4')
+
+
+class TestShowCases(BasePlanCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.request = RequestFactory()
+
+    def test_get_selected_testcases_works_with_string_pks(self):
+        """
+        Assures that tcms.testcases.views.get_selected_testcases
+        returns the same results, regardless of whether the
+        passed request contains the case pks as strings or
+        integers, as long as they are the same pks.
+        """
+
+        case_int_pks = [1, 2, 3, 4, 5, 6]
+        case_str_pks = []
+
+        for _pk in case_int_pks:
+            case_str_pks.append(str(_pk))
+
+        int_pk_query = self._get_selected_testcases(case_int_pks)
+        str_pk_query = self._get_selected_testcases(case_str_pks)
+
+        for case in TestCase.objects.filter(pk__in=case_int_pks):
+            self.assertTrue(case in int_pk_query and case in str_pk_query)
+
+    def _get_selected_testcases(self, pks):
+        return get_selected_testcases(self.request.post('tcms/testcases/', {'case': pks}))
```

Comment:
`tcms/testcases/` is not a valid URL path. Use the url path for some of the view functions which calls `get_selected_testcases()`. Use a `reverse()` instead of hard-coding the URL b/c the path can change in the future.